packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -657,6 +661,215 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
+      } else {
+        const oldValue = oldProps.value;
+        // Use Object.is to compare the new context value to the old value.
+        // Inlined Object.is polyfill.
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is
+        if (
+          (oldValue === newValue &&
+            (oldValue !== 0 || 1 / oldValue === 1 / newValue)) ||
+          (oldValue !== oldValue && newValue !== newValue) // eslint-disable-line no-self-compare
+        ) {
+          // No change.
+          changedBits = 0;
+        } else {
+          changedBits =
+            context.calculateChangedBits !== null
+              ? context.calculateChangedBits(oldValue, newValue)
+              : MAX_SIGNED_31_BIT_INT;
+          if (__DEV__) {
+            warning(
+              (changedBits & MAX_SIGNED_31_BIT_INT) === changedBits,
+              'calculateChangedBits: Expected the return value to be a ' +
+                '31-bit integer. Instead received: %s',
+              changedBits,
+            );
+          }
+          changedBits |= 0;
+
+          if (changedBits !== 0) {
+            propagateContextChange(
+              workInProgress,
+              context,
+              changedBits,
+              renderExpirationTime,
+            );
+          }
+        }
+      }
+
+      workInProgress.stateNode = changedBits;
+      pushProvider(workInProgress);
+
+      if (oldProps !== null && oldProps.children === newProps.children) {
+        return bailoutOnAlreadyFinishedWork(current, workInProgress);
+      }
+      const newChildren = newProps.children;
+      reconcileChildren(current, workInProgress, newChildren);
+      return workInProgress.child;
+    } else {
+      return null;
+    }
+  }
+
+  function updateContextConsumer(
+    current,
+    workInProgress,
+    renderExpirationTime,
+  ) {
+    if (enableNewContextAPI) {
+      const context: ReactContext<any> = workInProgress.type;
+      const newProps = workInProgress.pendingProps;
+
+      const newValue = context.currentValue;
+      const changedBits = context.changedBits;
+
+      if (changedBits !== 0) {
+        // Context change propagation stops at matching consumers, for time-
+        // slicing. Continue the propagation here.
+        propagateContextChange(
+          workInProgress,
+          context,
+          changedBits,
+          renderExpirationTime,
+        );
+      }
+
+      // Store the observedBits on the fiber's stateNode for quick access.
+      let observedBits = newProps.observedBits;
+      if (observedBits === undefined || observedBits === null) {
+        // Subscribe to all changes by default
+        observedBits = MAX_SIGNED_31_BIT_INT;
+      }
+      workInProgress.stateNode = observedBits;
+
+      const newChildren = newProps.render(newValue);
+      reconcileChildren(current, workInProgress, newChildren);
+      return workInProgress.child;
+    } else {
+      return null;
+    }
```

Comment:
I think the else in the end is unnecessary. You can simply return null. Also I'd suggest to refactor this into another functions because of readability.

Reply:
> I think the else in the end is unnecessary

This is just another code style, IMO, current implementation is for more readability

Reply:
testing